integration/check_examples_test.go

```diff
@@ -12,69 +15,145 @@ import (
 	"strings"
 	"testing"
 
+	"github.com/docker/docker/api/types/container"
+	"github.com/docker/docker/api/types/mount"
 	"github.com/stretchr/testify/assert"
 	"github.com/stretchr/testify/require"
+	"github.com/testcontainers/testcontainers-go"
+	"github.com/testcontainers/testcontainers-go/modules/registry"
 
+	"github.com/aquasecurity/trivy-checks/integration/testcontainer"
 	"github.com/aquasecurity/trivy-checks/internal/examples"
 	"github.com/aquasecurity/trivy/pkg/iac/framework"
 	"github.com/aquasecurity/trivy/pkg/iac/rego"
 	"github.com/aquasecurity/trivy/pkg/iac/rules"
 	"github.com/aquasecurity/trivy/pkg/types"
 )
 
-func TestValidateCheckExamples(t *testing.T) {
-	cacheDir := setupCache(t)
-	targetDir := setupTarget(t)
-	outputFile := filepath.Join(t.TempDir(), "report.json")
-
-	args := []string{
-		"conf",
-		"--skip-check-update",
-		"--quiet",
-		"--format", "json",
-		"--output", outputFile,
-		"--cache-dir", cacheDir,
-		targetDir,
-	}
-	runTrivy(t, args)
+var trivyVersions = []string{"0.57.1", "0.58.1", "latest", "canary"}
 
-	report := readTrivyReport(t, outputFile)
+func TestScanCheckExamples(t *testing.T) {
+	ctx := context.Background()
 
-	verifyExamples(t, report, targetDir)
-}
+	tmpDir, err := os.MkdirTemp("", "trivy-checks-examples-*")
+	require.NoError(t, err)
+	t.Cleanup(func() { os.RemoveAll(tmpDir) })
 
-func setupCache(t *testing.T) string {
-	t.Helper()
+	examplesPath := filepath.Join(tmpDir, "examples")
+	setupTarget(t, examplesPath)
 
-	cmd := exec.Command("make", "create-bundle")
-	cmd.Dir = ".."
-	require.NoError(t, cmd.Run())
-	defer os.Remove("../bundle.tar.gz")
+	targetDir, err := filepath.Abs(tmpDir)
+	require.NoError(t, err)
+
+	registryContainer, err := registry.Run(ctx, "registry:2")
+	require.NoError(t, err)
 
-	cacheDir := t.TempDir()
+	registryHost, err := registryContainer.HostAddress(ctx)
+	require.NoError(t, err)
 
-	policyDir := filepath.Join(cacheDir, "policy", "content")
-	require.NoError(t, os.MkdirAll(policyDir, os.ModePerm))
+	bundleImage := registryHost + "/" + "trivy-checks:latest"
+
+	bundlePath := buildBundle(t)
+	pushBundle(t, ctx, bundlePath, bundleImage)
+
+	for _, version := range trivyVersions {
+		t.Run(version, func(t *testing.T) {
+			reportFileName := version + "_" + "report.json"
+			args := []string{
+				"conf",
+				"--checks-bundle-repository", bundleImage,
+				"--format", "json",
+				"--output", "/testdata/" + reportFileName,
+				"--include-deprecated-checks=false",
+				"/testdata/examples",
+			}
+
+			trivy, err := testcontainer.RunTrivy(ctx, "aquasec/trivy:"+version, args,
```

Comment:
Do we need to call `Terminate()` on the testcontainers that we create to properly clean them up?